aya/src/programs/kprobe.rs

```diff
@@ -71,8 +71,12 @@ impl KProbe {
     /// target function.
     ///
     /// The returned value can be used to detach from the given function, see [KProbe::detach].
-    pub fn attach(&mut self, fn_name: &str, offset: u64) -> Result<KProbeLinkId, ProgramError> {
-        attach(&mut self.data, self.kind, Path::new(fn_name), offset, None)
+    pub fn attach<T: AsRef<Path>>(
```

Comment:
As per discord conversation, this is not a `Path`.

I think the most accurate and intuitive (for users) type would be `OsStr` and `AsRef<OsStr>` works for the common case `attach("foo")`. To work with bytes though you have to type `attach(OsStr::from_bytes(b"foo"))` since `OsStr` can only be created from bytes via `std::os::unix::ffi::OsStrExt`.

Therefore I'm leaning towards just making it `AsRef<[u8]>` for convenience.

Reply:
Thanks, I dug around a bit and came to the same conclusion. It looks like the meaning of this argument differs based on the type of probe. We could probably do better with the type safety of `ProbeKind` to encode this in the type system, but it seems correct. I've added a comment.